### PR TITLE
[WIP] Disallow moving literals to their own line in binary expressions

### PR DESCRIPTION
**Pull request by jlongster**
This is another proposed solution to #314, see the discussion there.

This also solves @llimllib case in #46 (see comment https://github.com/jlongster/prettier/issues/46#issuecomment-271696399). Currently we output his/her worst solution. Now we would output:

```js
lots +
  of +
  additions +
  are +
  put +
  into +
  some +
  weird +
  format +
  thats +
  ugly > 0;
```

While that might look a little weird with that example code, I think this will solve a lot of real-world problems where we'd move a simple literal to its own line.

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -2265,7 +2265,8 @@ function printBinaryishExpressions(path, parts, print) {
       parts.push(path.call(print, "left"));
     }
 
-    parts.push(" ", node.operator, line, path.call(print, "right"));
+    const allowLine = !namedTypes.Literal.check(node.right);
```
Comment (jlongster) [inline src/printer.js]: I just remembered that doing this type of check is better than `node.type === "Literal"`. It handles different across parsers better, i.e. this works on both and babylon even though they handle the literal types differently.

Comment (vjeux) [inline src/printer.js]: Nice!

**Comment by vjeux** [comment]
Can you add a test as well, this way if someone is going to tweak that logic in the future, it's going to show a difference.

**Comment by jlongster** [comment]
Oh definitely. I was mainly opening this as a proof of concept to see what people though. Add tests though would be a good way to show the difference.

**Comment by jlongster** [comment]
Do want identifiers to be included in this test too? Currently we might still break them onto newlines, but I think that might be OK.

Also, this PR doesn't quite work. Now it doesn't break any of the expression at all if it's all made up of numbers (if we check idents too, it'll break even more). This is because even the sub-expressions have a literal/ident on the right-hand side too. I suppose we should check if the left-hand is *not* a binary expression too, which would only trigger them when it's a simpler expression like `ident + 5`.

**Comment by jlongster** [comment]
Considering closing this since we have another solution in #314, or does anything thing we should still also implement this?

**Comment by jlongster** [comment]
Closing this for now, we can reopen if later we find other scenarios where it's helpful. Let's get #314 in first and see how well that works.

**Pull request closed without merging**